sdk/data/azcosmos/queryengine/cosmos_query_engine.go

```diff
@@ -13,15 +13,36 @@ type QueryEngine interface {
 type QueryRequest struct {
 	// PartitionKeyRangeID is the ID of the partition key range from which data is requested.
 	PartitionKeyRangeID string
+	// The index of this request, within the partition key range.
+	//
+	// This value will always increase for subsequent requests for the same partition key range.
+	// It must be provided back to the pipeline when providing data, so that the pipeline can ensure that data is provided in order.
+	Index uint
 	// Continuation is the continuation token to use in the request.
 	Continuation string
+	// The query to execute for this partition key range, if different from the original query.
+	Query string
+	// If a query is specified, this flag indicates if the query parameters should be included with that query.
+	//
+	// Sometimes, when an override query is specified, it differs in structure from the original query, and the original parameters are not valid.
+	IncludeParameters bool
+	// If specified, indicates that the SDK should IMMEDIATELY drain all remaining results from this partition key range, following continuation tokens, until no more results are available.
+	// All the data from this partition key range should be provided BEFORE any new items will be made available.
+	//
+	// This allows engines to optimize for non-streaming scenarios, where the entire result set must be provided to the engine before it can make progress.
+	Drain bool
 }
 
 // QueryResult contains the result of a query for a specific partition key range.
 type QueryResult struct {
+	// The ID of the partition key range that was queried.
 	PartitionKeyRangeID string
-	NextContinuation    string
-	Data                []byte
+	// The index of the QueryRequest that generated this result.
+	RequestIndex uint
```

Comment:
Similar to the above, this becomes `RequestId` (and the doc comment should be updated similarly.